File: IttFelTeheted-SPA/src/app/_models/post.ts
import { Topic } from './topic';
import { Answer } from './answer';
import { Photo } from './photo';

export interface Post {
    id: number;
    title: string;
    dateAdded: Date;
    views: number;
    answerCount: number;
    userPhotoUrl?: string;
    postBody?: string;
    userId?: number;
    topic?: Topic[];
    userName?: string;
    photoUrl?: string;
    answers?: Answer[];
    photos?: Photo[];
}

File: IttFelTeheted.API/Models/Post.cs
using System;
using System.Collections.Generic;

namespace IttFelTeheted.API.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string PostBody { get; set; }
        public User User { get; set; }
        public virtual Topic Topic { get; set; }
        public DateTime DateAdded { get; set; }
        public int Views { get; set; }
        public string PhotoUrl { get; set; }
        public ICollection<Answer> Answers { get; set; }

    }
}